### PR TITLE
Add documentataion for DATEPARSE

### PR DESCRIPTION
**Pull request by jainam1995**
Add a section under the date-functions for dateparse as it is a little different from the other usual date functions.

Kept it a little short, like the other paras on the page and also linked an example of dateparse from Annoteted.tdd file for now . We may want to re-link it to the postgres_odbc or postgres_jdbc that will be added in future if required. Also tried to explain a little based off the example so its better to understand,

### CODE REVIEW

**Comment by rosswbrown** [comment]
Overall content is good.  Let's wait for #525 to go in and merge styles a bit and re-review this.

**Comment by jainam1995** [comment]
> Overall content is good. Let's wait for #525 to go in and merge styles a bit and re-review this.

Update the format of the subsection DATEPARSE function to match the one in #525

File: docs/dialect.md [hunk 1]
```diff
@@ -113,6 +113,11 @@ The function <span style="font-family: courier new">name</span> must be one of t
...
+  As an example, for the [DATEPARSE function](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/Annotated.tdd#L945) with arguments <span style="font-family: courier new">%1</span> and <span style="font-family: courier new">%2</span>, 
+  the string values for <span style="font-family: courier new">%1</span> are defined by [icu-token-map](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/Annotated.tdd#L1369) and the string values for <span style="font-family: courier new">%2</span> are defined by [date-literal-escape](https://github.com/tableau/connector-plugin-sdk/blob/master/samples/components/dialects/Annotated.tdd#L1130) .
 
   
   Like <span style="font-family: courier new">function</span>, <span style="font-family: courier new">date-function</span> requires name and return-type, but unlike <span style="font-family: courier new">function</span>, group is not required.
```
Comment (rosswbrown) [inline docs/dialect.md:123]: Let's move this line about return type and group up to before Example: DATEPART

Comment (jainam1995) [reply docs/dialect.md]: Fixed

**Pull request merged**